src/app/qgsmeasuretool.cpp

```diff
@@ -125,7 +126,37 @@ void QgsMeasureTool::updateSettings()
   mRubberBandPoints->setIcon( QgsRubberBand::ICON_CIRCLE );
   mRubberBandPoints->setIconSize( 10 );
   mRubberBandPoints->setColor( QColor( myRed, myGreen, myBlue, 150 ) );
+
+  // Reproject the points to the new destination CoordinateReferenceSystem
+  if ( mRubberBand->size() > 0 && mDestinationCrs != mCanvas->mapSettings().destinationCrs() )
+  {
+    QList<QgsPoint> points = mPoints;
+    bool lastDone = mDone;
+    int index = 0;
+
+    mDialog->restart();
+    mDone = lastDone;
+    QgsCoordinateTransform ct( mDestinationCrs, mCanvas->mapSettings().destinationCrs() );
+
+    Q_FOREACH ( const QgsPoint& previousPoint, points )
+    {
+      QgsPoint point = ct.transform( previousPoint );
```

Comment:
Sorry - one last thing. You should probably catch any QgsCsExceptions here and skip the point